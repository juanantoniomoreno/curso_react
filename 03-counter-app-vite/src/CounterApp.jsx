import { useState } from 'react'
import PropTypes from 'prop-types'
import { Button } from 'react-bootstrap'

export const CounterApp = ({ value }) => {

  const [ counter, setCounter ] = useState( value );
  
  const handleAdd = () => { 
    
    setCounter( counter + 1 );
    //setCounter( (c) => c + 1  )
  }

  const handleSubstract = () => {
    setCounter( counter - 1 );
  }

  const handleReset = () => {
    setCounter( value );
  }

  return (
    <>
        <h1>CounterApp</h1>
        <h2>{ counter }</h2>
        <div>
          <Button 
            variant='info'
            onClick={ handleAdd }
            >+1
          </Button>
          <Button 
            variant='warning'
            onClick={ handleSubstract }
            > -1
          </Button>
          <Button 
            variant='secondary'
            onClick={ handleReset }
            aria-label = "btn-reset"
            >Reset
          </Button>
        </div>
    </>
  )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value: 0
}


