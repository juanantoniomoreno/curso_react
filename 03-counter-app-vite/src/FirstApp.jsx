import PropTypes from 'prop-types';

const getSaludo = (name) => {
  return `Hola ${name}`;
}

const getAge = (birth_year) => {
  let now = new Date();
  return (now.getFullYear() - birth_year);
}

export const FirstApp = ({ title, subTitle, name }) => {

  //console.log(props);
      
  return (
    <>
      <h1 data-testid="test-title" > { title }</h1>
      <p>{ subTitle }</p>
      <p>{ subTitle }</p>
      {/* <h3>Tengo { subTitle } años</h3> */}
      {/* <code>{ JSON.stringify( newMsg ) }</code> */}
      <p>{ name }</p>
    </>
  )
}

//Para hacer que las props sean obligatorias y sean de un tipo determinado
FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired
}

//Podemos dar valores por defecto a las props
FirstApp.defaultProps = {
  /* title: 'No hay título', */
  subTitle: 'No hay subtítulo',
  name: 'Juan'
}
