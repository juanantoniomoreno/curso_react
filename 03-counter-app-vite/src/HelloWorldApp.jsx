import React from 'react'
import { FirstComponent } from './FirstComponent'

export const HelloWorldApp = () => {
  return (
    <div>
      Hola Mundo!!
    </div>
  )
}