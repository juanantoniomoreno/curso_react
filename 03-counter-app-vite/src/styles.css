html, body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 50px;
}

button {
    margin: 0.1rem;
}