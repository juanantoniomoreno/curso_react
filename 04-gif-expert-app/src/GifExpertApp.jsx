import { useState } from 'react';
import { AddCategory } from './components/AddCategory';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch', 'Dragon Ball']);

    const onAddCategory = () => {

        setCategories([ 'Valorant', ...categories ])
    }

    console.log( categories );
    
    

  return (
    <>
        {/* *TITULO */}
        <h1>GifExpertApp</h1>

        {/* INPUT */}
        <AddCategory />

        {/* LISTADO DE GIFT */}
        <button onClick={ onAddCategory }>Agregar</button>
        <ol>
            {categories?.map( (category, index) => {
                return(
                    <li key={ index }>{category}</li>
                )
            })}
        </ol>

    </>
  )
}
