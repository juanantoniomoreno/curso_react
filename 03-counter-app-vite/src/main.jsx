import React from 'react'
import ReactDOM from 'react-dom/client';
import 'bootstrap/dist/css/bootstrap.min.css';
import { CounterApp } from './CounterApp';
/* import { FirstApp } from './FirstApp'; */

import './styles.css';

ReactDOM.createRoot( document.getElementById('root')).render(
    <React.StrictMode>
        {/* <FirstApp title='Hola, Soy Vegeta'/> */}
        <CounterApp value={ 0 }/>
    </React.StrictMode>
);