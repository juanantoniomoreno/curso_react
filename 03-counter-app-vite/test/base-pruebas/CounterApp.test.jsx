import { fireEvent, render, screen } from "@testing-library/react"
import { CounterApp } from "../../src/CounterApp"


describe('Pruebas en el <Counter App />', () => {

    const initialValue = 10;

    test('Hacer match con el Snapshot', ()=> {

        const { container } = render(<CounterApp value={ initialValue } />);
        expect( container ).toMatchSnapshot();

    })

    test('Debe mostrar el valor inicial de 100 <CounterApp value={100}', ()=>{

        render( <CounterApp value={ 100 } />);
        expect( screen.getByText(100) ).toBeTruthy();
    })

    test('Debe incrementar con el botón +1', ()=> {

        render( <CounterApp value={ initialValue } />);
        fireEvent.click( screen.getByText('+1') )
        expect( screen.getByText( '11' )).toBeTruthy();
    })

    test('Debe decrementar con el botón -1', ()=> {

        render( <CounterApp value={ initialValue } />);
        fireEvent.click( screen.getByText('-1') )
        //screen.debug()
        expect( screen.getByText( '9' )).toBeTruthy();
    })

    test('Debe funcionar el botón de Reset', () => {
        
        render( <CounterApp value={ 10 } />);
        fireEvent.click( screen.getByText( '+1' ) );
        fireEvent.click( screen.getByText( '+1' ) );
        //fireEvent.click( screen.getByText( 'Reset' ));

        fireEvent.click( screen.getByRole('button', { name: 'btn-reset' }) );

        expect( screen.getByText( 10 )).toBeTruthy();
    })


})