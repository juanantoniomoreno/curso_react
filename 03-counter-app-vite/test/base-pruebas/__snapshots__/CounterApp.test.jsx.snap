// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <Counter App /> Hacer match con el Snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    10
  </h2>
  <div>
    <button
      class="btn btn-info"
      type="button"
    >
      +1
    </button>
    <button
      class="btn btn-warning"
      type="button"
    >
       -1
    </button>
    <button
      aria-label="btn-reset"
      class="btn btn-secondary"
      type="button"
    >
      Reset
    </button>
  </div>
</div>
`;
